Project-Folder/SSS/app/src/main/java/coen390/nicholas/sss/testActivity.java

```diff
@@ -72,27 +96,75 @@ private void setText()
 
     private void setQuestion()
     {
-        String theQuestion = "Question undefined";
+        getQuestion();
+        String ask = "Sign the Following: " + theQuestion;
+
+        question.setText(ask);
+    }
+
+    private void getQuestion()
+    {
+        Random rndIndex = new Random();
+        Boolean foundNew = false;
+        int qIndex = 0;
+
         if (level == 1) {
-            theQuestion = "Sign the Following: " + quizTracking.getQuestion(tutorialActivitiy.getCurrent(level));
+            theQuestion = quizTracking.getQuestion(tutorialActivitiy.getCurrent(level));
         }
         else if (level == 2){
-            theQuestion = "Sign the Following: " + quizTracking.getQuestion(2 * tutorialActivitiy.getCurrent(level));
+            while(!foundNew) { //loop until you find a new question index
+                qIndex = rndIndex.nextInt(26); //get a new random index from 0 to 25
+                if (tutorialActivitiy.getCurrent(level) != 0) {
+                    for (int i = 0; i < tutorialActivitiy.getCurrent(level); i++) { //check if it exist
+                        if (allQuestions[i] != qIndex) //if it doesnt
+                        {
+                            foundNew = true;
+                        } else {
+                            foundNew = false;
```

Comment:
you unset the fact that you found a duplicate

Reply:
I should break when the condition is assigned false right? So I break out of the loop when I find a duplicate

Reply:
in the if statement add the break... u want to stop looking ones u found it

Reply:
I’m a bit confused because in the if I set it to true only if it’s unique. But I want to make sure it’s unique in all of the previous (hence the for loop). Shouldn’t I break when I found a duplicate which occurs in the else because when you find a duplicate, the new index isn’t valid and I must look for a new one??

Reply:
you understand your code best that's not what thought `foundNew` was but I think you understand where your issue may lie, try both it's gonna be one of them

Reply:
it was actually neither. I tried both but it still gives duplicates

Reply:
i ended up changing the condition for the if